File: simple-stack-example-basic-java-fragment/src/main/java/com/zhuinden/navigationexamplefrag/screens/NotificationKey.java
package com.zhuinden.navigationexamplefrag.screens;

import com.google.auto.value.AutoValue;
import com.zhuinden.navigationexamplefrag.core.navigation.BaseFragment;
import com.zhuinden.navigationexamplefrag.core.navigation.BaseKey;

/**
 * Created by Owner on 2017. 06. 29..
 */
@AutoValue
public abstract class NotificationKey extends BaseKey {
    public static NotificationKey create() {
        return new AutoValue_NotificationKey();
    }

    @Override
    protected BaseFragment createFragment() {
        return new NotificationFragment();
    }
}

File: examples/simple-stack-example-basic-kotlin-fragment/src/main/java/com/zhuinden/simplestackexamplekotlinfragment/screens/NotificationKey.kt
package com.zhuinden.simplestackexamplekotlinfragment.screens

import com.zhuinden.simplestackexamplekotlinfragment.core.navigation.BaseFragment
import com.zhuinden.simplestackexamplekotlinfragment.core.navigation.BaseKey
import kotlinx.android.parcel.Parcelize

/**
 * Created by Owner on 2017.11.13.
 */
@Parcelize
data class NotificationKey(private val placeholder: String = "") : BaseKey() { // generate reliable `toString()` for no-args data class
    override fun createFragment(): BaseFragment = NotificationFragment()
}
